src/sentry/snuba/discover.py

```diff
@@ -355,6 +358,16 @@ def query(
     # Resolve the public aliases into the discover dataset names.
     snuba_args, translated_columns = resolve_discover_aliases(snuba_args)
 
+    # Make sure that any aggregate conditions are also in the selected columns
+    for having_clause in snuba_args.get("having"):
+        found = any(
+            having_clause[0] == agg_clause[-1] for agg_clause in snuba_args.get("aggregations")
+        )
+        if not found:
+            raise InvalidSearchQuery(
+                "Aggregates used in a condition must also be in the selected columns."
```

Comment:
Should this include which aggregate wasn't found in the selected fields so the user can sort out what to do next more easily?

Reply:
I prefer to make things as difficult and opaque for the user as possible.

I didn't do that because at this point we only have the formatted column names (`max_time`) instead of what they actually put in the query (`max(time)`). I thought it might be more confusing to have those in the error message. What do you think?

Reply:
Good point about the formatted aliases being confusing.